File: app/src/main/java/com/example/administrator/Item/Mall.java
package com.example.administrator.Item;


import org.dom4j.Element;

import java.util.HashMap;

public class Mall extends Item {

    public Mall(String name, int volume, int originalPrice, int total) {
        super(name, volume, originalPrice, total);
    }

    @Override
    public void purchase() {

    }

    @Override
    public void getXMLDate(Element element) {

    }

    @Override
    public void getDate(String name, String ItemName) {

    }

    @Override
    public void saveDate(String tableName) {

    }

    @Override
    public void getDate(HashMap<String, Item> articles) {

    }
}

File: app/src/main/java/com/example/administrator/item/Mall.kt
package com.example.administrator.item

import android.database.Cursor

class Mall : Item {

    lateinit var list: Array<Item>

    constructor(name: String, volume: Int, originalPrice: Int, itemList: Array<Item>) : super(name, volume, originalPrice) {
        list = itemList
    }

    constructor() : super()

    override fun haveTable(): Boolean {
        return false
    }

    override fun createTable(name: String) {}

    override fun getSQLDate(cursor: Cursor) {}

    override fun saveDate(workSpaceName: String) {}
    override fun<T: Item> getListItem(name: String): T {
        Item.changeToMap(items)[name]?.let { return it as T}
        error("Item No Found")
    }


    companion object {

        var items = arrayOf<Mall>()
    }

}